api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupMERSemiAnnualReport.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.eptsreports.reporting.reports;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import org.openmrs.module.eptsreports.reporting.library.cohorts.TXTBCohortQueries;
+import org.openmrs.module.eptsreports.reporting.library.datasets.TxTBDataset;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils;
+import org.openmrs.module.reporting.ReportingException;
+import org.openmrs.module.reporting.evaluation.parameter.Mapped;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SetupMERSemiAnnualReport extends SetupMERQuarterly {
+
+  @Autowired private TxTBDataset txTBDataset;
+  @Autowired private TXTBCohortQueries txtBCohortQueries;
+
+  @Override
+  public String getVersion() {
+    return "1.0-SNAPSHOT";
+  }
+
+  @Override
+  public String getUuid() {
+    return "00812f4a-af74-4227-8698-d46cd7a21ef0";
+  }
+
+  @Override
+  public String getExcelDesignUuid() {
+    return "00812f4a-af74-4227-8698-d46cd7a21ef1";
+  }
+
+  @Override
+  public String getName() {
+    return "MER Semi Annual";
+  }
+
+  @Override
+  public String getDescription() {
+    return "MERSemiAnnual - PEPFAR MER 2.1";
+  }
+
+  @Override
+  public ReportDefinition constructReportDefinition() {
+    ReportDefinition reportDefinition = super.constructReportDefinition();
+
+    reportDefinition.setBaseCohortDefinition(
```

Comment:
@kaweesi Shouldn't we be using `genericCohortQueries.getBaseCohort()` as base cohort?

Reply:
are these 2 doing the same, probably using that base cohort would have broken my former matching figures with existing report, glad we are on changes

Reply:
That is the base cohort we are using on all reports. It comes from the SQL reports from FGH. I think you should use it on this code.

Reply:
because of the changes we are already working on, i changed this, i kind of think this change may have affected the old report had it been a match with old numbers that we were considering, am however not sure if all reports had the same base cohort before, if so then probably that wouldn't have affected the matchings